include/RAJA/util/SubView.hpp

```diff
@@ -0,0 +1,268 @@
+ */
+
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+// Copyright (c) 2016-25, Lawrence Livermore National Security, LLC
+// and RAJA project contributors. See the RAJA/LICENSE file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+
+#ifndef RAJA_SUBVIEW_HPP
+#define RAJA_SUBVIEW_HPP
+
+#include "RAJA/util/for_each.hpp"
+#include "RAJA/util/macros.hpp"
+#include "RAJA/util/types.hpp"
+#include "camp/tuple.hpp"
+#include "camp/array.hpp"
+
+namespace RAJA
+{
+
+template<typename IndexType = Index_type>
+struct RangeSlice {
+    IndexType start_, end_; 
+
+    static constexpr bool reduces_dimension = false;
+
+    RAJA_INLINE RAJA_HOST_DEVICE constexpr IndexType map_index(const IndexType& idx) const {
+        return start_ + idx;
+    }
+
+    template<IndexType DIM, typename LayoutType>
+    RAJA_INLINE RAJA_HOST_DEVICE constexpr IndexType size(const LayoutType&) const {
+        return (end_ - start_ + 1);
+    }
+
+    RAJA_INLINE RAJA_HOST_DEVICE constexpr IndexType stride() const {
+        return 1;
+    }
+};
+
+template<typename IndexType = Index_type>
+struct RangeStartSlice {
+    IndexType start_; 
+
+    static constexpr bool reduces_dimension = false;
+
+    RAJA_INLINE RAJA_HOST_DEVICE constexpr IndexType map_index(const IndexType& idx) const {
+        return start_ + idx;
+    }
+
+    template<IndexType DIM, typename LayoutType>
+    RAJA_INLINE RAJA_HOST_DEVICE constexpr IndexType size(const LayoutType& layout) const {
+        return (layout.size() - start_);
+    }
+
+    RAJA_INLINE RAJA_HOST_DEVICE constexpr IndexType stride() const {
+        return 1;
+    }
+};
+
+template<typename IndexType = Index_type>
+struct FixedSlice { 
+    IndexType idx_; 
+
+    static constexpr bool reduces_dimension = true;
+
+    RAJA_INLINE RAJA_HOST_DEVICE constexpr IndexType map_index(const IndexType&) const {
+        return idx_;
+    }
+
+    template<IndexType DIM, typename LayoutType>
+    RAJA_INLINE RAJA_HOST_DEVICE constexpr IndexType size(const LayoutType&) const {
+        return 1;
+    }
+
+    RAJA_INLINE RAJA_HOST_DEVICE constexpr IndexType stride() const {
+        return 1;
+    }
+
+};
+
+template<typename IndexType = Index_type>
+struct NoSlice { 
+    static constexpr bool reduces_dimension = false;
+
+    RAJA_INLINE RAJA_HOST_DEVICE constexpr IndexType map_index(const IndexType& idx) const {
+        return idx;
+    }
+
+    template<IndexType DIM, typename LayoutType>
+    RAJA_INLINE RAJA_HOST_DEVICE constexpr IndexType size(const LayoutType& layout) const {
+        return layout.template get_dim_size<DIM>();
+    }
+
+    RAJA_INLINE RAJA_HOST_DEVICE constexpr IndexType stride() const {
+        return 1;
+    }
+};
+
+template<typename IndexType = Index_type>
+struct StridedSlice { 
+    IndexType start_, end_, stride_;
```

Comment:
Does this support negative strides? I'm pretty sure that the strided range supports negative strides.